File: python/2.7/factory/CheesePizza.py
from Pizza import *

class CheesePizza(Pizza):
    def __init__(self, ingredientFactory):
        super(CheesePizza, self).__init__()
        self.ingredientFactory = ingredientFactory

    def prepare(self):
        print "Preparing %s" %(self.name)
        self.dough = self.ingredientFactory.createDough()
        self.sauce = self.ingredientFactory.createSauce()
        self.cheese = self.ingredientFactory.createCheese()

File: js/node/factory/pizza/CheesePizza.js
require('./Pizza');

CheesePizza = function(ingredientFactory) {
    Pizza.apply(this);
    this.ingredientFactory = ingredientFactory;
}
CheesePizza.prototype = new Pizza();
CheesePizza.prototype.prepare = function() {
    console.log("Prepareing " + this.name);
    this.dough = this.ingredientFactory.createDough();
    this.sauce = this.ingredientFactory.createSauce();
    this.cheese = this.ingredientFactory.createCheese();
}